File: src/algorithms/contains-duplicate.js
/**
 * Given an integer array nums, returns true if any value appears at least
 * twice in the array, and returns false if every element is distinct.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {boolean} - Returns true if all elements are distinct
 */
function containsDuplicate (nums) {
  let elements = new Set();
  for (let i = 0; i < nums.length; i++) {
    if (elements.has(nums[i])) {
      return true;
    } else {
      elements.add(nums[i]);
    }
  }
  return false;
}

File: src/algorithms/contains-duplicate.py
def contains_duplicate(nums):
    """Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

    Args:
        nums (List[int]): A list of integers

    Returns:
        Boolean: Returns true any value in nums appears at least twice
    """
    seen = set()
    for i in range(len(nums)):
        if nums[i] in seen:
            return True
        else:
            seen.add(nums[i])
    return False